modules/cabal-project.nix

```diff
@@ -90,7 +90,7 @@ in {
       '';
     };
     sha256map = mkOption {
-      type = nullOr unspecified;
+      type = nullOr (attrsOf (attrsOf str));
```

Comment:
When using this for a `repository` there is is no `tag`  https://github.com/input-output-hk/haskell.nix/blob/2b9b99a537327df4dfaa93d7827394c5388bceb7/lib/call-cabal-project-to-nix.nix#L37-L39

So this type probably needs to be:
```
type = nullOr (attrsOf (either str (attrsOf str)));
```

Reply:
These should have been separate arguments or somehow have had the same shape. Union types are in general a bad idea. Will fix this though.

Reply:
Agreed.  `inputMap` has the same problem.

We could have a single level and use `"${url}#${tag}"` and `"${url}"` as the keys.